plugin/trino-hive/src/main/java/io/trino/plugin/hive/TableToPartitionMapping.java

```diff
@@ -106,7 +111,7 @@ public int getEstimatedSizeInBytes()
                 estimatedSizeOf(partitionColumnCoercions, (Integer key) -> INTEGER_INSTANCE_SIZE, HiveTypeName::getEstimatedSizeInBytes) +
                 OPTIONAL_INSTANCE_SIZE +
                 tableToPartitionColumns
-                        .map(tableToPartitionColumns -> estimatedSizeOf(tableToPartitionColumns, (Integer key) -> INTEGER_INSTANCE_SIZE, (Integer value) -> INTEGER_INSTANCE_SIZE))
+                        .map(tableToPartitionColumns -> INT_2_INT_ARRAY_MAP_INSTANCE_SIZE + 2 * sizeOfIntArray(tableToPartitionColumns.size()))
```

Comment:
let's change the type of the tableToPartitionColumns field to be Optional<Int2IntArrayMap> 

let's also add a rationale for this.
Int2IntArrayMap is less typically used for "compression."

Reply:
It is Optional actually.

Added rationale